lib/config/default.js

```diff
@@ -1223,18 +1228,11 @@ function buildConfig(definition, config, paths = [], objectKeys = 1) {
     const type = typeof value
     keysSeen++
     if (type === 'string') {
-      if (paths.length) {
-        setNestedKey(conf, [...paths, key], value)
-      } else {
-        conf[key] = value
-      }
+      assignConfigValue({ config: conf, key, value, paths })
     } else if (type === 'object') {
       if (value.hasOwnProperty('default')) {
-        if (paths.length) {
-          setNestedKey(conf, [...paths, key], value.default)
-        } else {
-          conf[key] = value.default
-        }
+        assignConfigValue({ config: conf, key, value: value.default, paths })
+        // add the current leaf node key to the paths and recurse through function again
```

Comment:
can we move this down to inside the else block? I think this is what the comment is referring to, since that's where we call `buildConfig` again

Reply:
yea prettier fixed this and it looks worse now.